llama_index/vector_stores/pinecone.py

```diff
@@ -265,8 +277,14 @@ def from_params(
         except ImportError:
             raise ImportError(import_err_msg)
 
-        pinecone.init(api_key=api_key, environment=environment)
-        pinecone_index = pinecone.Index(index_name)
+        pinecone_client_version = get_distribution("pinecone-client").version
+
+        if version.parse(pinecone_client_version) >= version.parse("3.0.0"):
```

Comment:
NIT: I think we should explain in comments perhaps above this whole block that after version 3.0.0 of the Pinecone client, we've make a breaking change to the initialization signature, hence this code.

Reply:
done

Reply:
Done